File: Kattis/Luhnchecksum.java
import java.util.Scanner;

public class Luhnchecksum{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int tests = Integer.parseInt(in.nextLine());

        for(int t = 0; t < tests; t++){            
            StringBuilder s = new StringBuilder(in.nextLine()).reverse();
            int[] iarr = new int[s.length()];
            for(int i = 0; i < iarr.length; i++){
                iarr[i] = Character.getNumericValue(s.charAt(i));
            }
            int sum = 0;
            for (int c = 0; c < iarr.length; c++){
                int n = iarr[c];
                if(c % 2 == 1){
                    n *= 2;
                    if (n > 9){
                        String ns = Integer.toString(n) ;
                        n = Character.getNumericValue(ns.charAt(0)) + Character.getNumericValue(ns.charAt(1));
                    }
                }
                sum += n;
            }
            
            if(sum % 10 == 0){
                System.out.println("PASS");
            } else {
                System.out.println("FAIL");
            }
            
        }
    }
}

File: Kattis/Luhnchecksum.py
tests = int(input())

for i in range(tests):
    s = input()[::-1]
    sum = 0
    for i in range(len(s)):
        inti = int(s[i])
        if i % 2 == 1:
            inti = inti * 2
            if inti > 9:
                sint = str(inti)
                inti = int(sint[0]) + int(sint[1])
        sum += inti
    if sum % 10 == 0:
        print("PASS")
    else:
        print("FAIL")
